ui/redux/actions/memberships.js

```diff
@@ -0,0 +1,377 @@
+import { selectFetchingIdsForMembershipChannelId, selectIsListingAllMyTiers } from 'redux/selectors/memberships';
+import { selectChannelTitleForUri, selectMyChannelClaims } from 'redux/selectors/claims';
+import { doOpenModal } from 'redux/actions/app';
+import { ODYSEE_CHANNEL } from 'constants/channels';
+import { formatDateToMonthDayAndYear } from 'util/time';
+import { buildURI } from 'util/lbryURI';
+
+import { getStripeEnvironment } from 'util/stripe';
+const stripeEnvironment = getStripeEnvironment();
+
+export const doFetchChannelMembershipsForChannelIds = (channelId: string, channelIds: ClaimIds) => async (
+  dispatch: Dispatch,
+  getState: GetState
+) => {
+  if (!channelIds || channelIds.length === 0) return;
+
+  // remove dupes and falsey values
+  const dedupedChannelIds = [...new Set(channelIds)].filter(Boolean);
+
+  // check if channel id is fetching
+  const state = getState();
+  const fetchingForChannel = selectFetchingIdsForMembershipChannelId(state, channelId);
+  const fetchingSet = fetchingForChannel && new Set(fetchingForChannel);
+
+  const channelsToFetch = dedupedChannelIds.filter((dedupedChannelId) => {
+    const notFetching = !fetchingSet || !fetchingSet.has(dedupedChannelId);
+    return notFetching;
+  });
+
+  if (channelsToFetch.length === 0) return;
+
+  // create 'comma separated values' string for backend
+  const channelIdsToFetch = channelsToFetch.join(',');
+
+  dispatch({ type: ACTIONS.CHANNEL_MEMBERSHIP_CHECK_STARTED, data: { channel: channelId, ids: channelsToFetch } });
+
+  return await Lbryio.call('membership', 'check', {
+    channel_id: channelId,
+    claim_ids: channelIdsToFetch,
+    environment: stripeEnvironment,
+  })
+    .then((response) => {
+      const membershipsById = {};
+
+      for (const channelId in response) {
+        const memberships = response[channelId];
+
+        // if array was returned for a user (indicating a membership exists), otherwise is null
+        if (Number.isInteger(memberships?.length)) {
+          for (const membership of memberships) {
+            if (membership.activated) {
+              membershipsById[channelId] = membership.name;
+            }
+          }
+        }
+
+        if (!membershipsById[channelId]) membershipsById[channelId] = null;
+      }
+
+      return dispatch({ type: ACTIONS.CHANNEL_MEMBERSHIP_CHECK_COMPLETED, data: { channelId, membershipsById } });
+    })
+    .catch((e) => dispatch({ type: ACTIONS.CHANNEL_MEMBERSHIP_CHECK_FAILED, data: { channelId, error: e } }));
+};
+
+export const doFetchOdyseeMembershipForChannelIds = (channelIds: ClaimIds) => async (dispatch: Dispatch) =>
+  dispatch(doFetchChannelMembershipsForChannelIds(ODYSEE_CHANNEL.ID, channelIds));
+
+export const doMembershipList = (params: MembershipListParams) => async (dispatch: Dispatch) =>
+  await Lbryio.call('membership', 'list', { environment: stripeEnvironment, ...params }, 'post')
+    .then((response: MembershipTiers) =>
```

Comment:
:small_red_triangle: A bunch of calls with bad params in Rabbit Hole

<img width="1279" alt="image" src="https://user-images.githubusercontent.com/64950861/191524532-47f29583-2a1b-4d87-9664-60f0a3902cd9.png">

Reply:
No longer exists ("fixed"), since the Join button is now restricted to channels and not every claim.